### ISSUE TITLE
Support @(Transactional)EventListener out of the box

### ISSUE THREAD

**Issue by odrotbohm** [spring-projects-experimental/spring-native#828]
Methods annotated with `@(Transactional)EventListener` currently don't work without custom hint declarations. They shouldn't need that. Here's what's currently needed to make them work:

```java
@TypeHint(types = { EventListener.class, TransactionalEventListener.class }, access = AccessBits.ANNOTATION)
@JdkProxyHint(types = {TransactionalEventListener.class, SynthesizedAnnotation.class })
```

It would be cool if the AOT component processing automatically registered the necessary information.

**Comment by sdeleuze** [comment]
@aclement I see [`Type#isEventListener()`](https://github.com/spring-projects-experimental/spring-native/blob/main/spring-aot/src/main/java/org/springframework/nativex/type/Type.java#L719) is not used anymore, any thoughts?

**Comment by aclement** [comment]
I created a new sample for events that is using `@EventListener` and `@TransactionalEventListener`. I've created a new `TransactionalEventListenerComponentProcessor` that will add the info @odrotbohm referred to if the annotation is detected. As part of this I also had to revisit the TransactionalComponentProcessor and fixed that up to do the right thing for transactional usage on a class (it previously had a very unreliable hack in that simply tries to create a JDK proxy for the types the transactional marked class was extending). This probably addresses the `@Transactional` comment in the other issue.
